src/Ombi/ClientApp/app/requests/tvrequests.component.ts

```diff
@@ -223,6 +222,22 @@ export class TvRequestsComponent implements OnInit {
         this.loadInit();
     }
 
+    private setDefaults(val: any) {
+        if (val.data.posterPath === null) {
+            val.data.posterPath = "../../../images/default_tv_poster.png";
+        }
+
+        if (val.data.imdbId.length === 0) {
+            this.searchService.getShowInformationTreeNode(val.data.tvDbId).subscribe(x => {
```

Comment:
In search results the TreeNode data has `seriesId` attribute which can be used with the url: `https://www.tvmaze.com/shows/{seriesId}` in order to display titles with urls in the event that that tvmaze data does not contain an `imdbId` for the particular show. 

However inside the requests page the TreeNode data does not have a `seriesId` attribute, so the only way I thought of to get this since we do have the `tvDbId` is to use the search service call to obtain this very reliably and this way tv show titles don't just link to a 404 imdb page. 

What do you think is a better approach? I mean we can set the `imdbId` attribute accordingly during search so we don't have to do this and this would work for new requests but then I was just thinking of the existing requests...

Reply:
Well we do try and set the IMDb Id, but we can't always get it because it's not available sometimes. What I'd recommend is if we dont have it then we either create a new api and get it OR get it on the requests API that returns the requested items.

Reply:
That's fine! We just need a new API assembly like all the other external API's.

Should be pretty simple to do.